util/src/main/java/com/coderanch/util/cli/InputUtility.java

```diff
@@ -0,0 +1,277 @@
+            }
+        }
+        return num.getAsDouble();
+    }
+
+    /**
+     * Tries to parse a Double from the string and validate it.
+     *
+     * @param line            the string to parse.
+     * @param doublePredicate the predicate used to validate the Integer.
+     * @return a validated Double or nothing.
+     */
+    private OptionalDouble tryDoubleParse(String line, Predicate<Double> doublePredicate) {
+        try {
+            var num = Double.parseDouble(line);
+            return doublePredicate.test(num) ? OptionalDouble.of(num) : OptionalDouble.empty();
+        }
+        catch (NumberFormatException e) {
+            return OptionalDouble.empty();
+        }
+    }
+
+    /**
+     * Gets the next yes or no answer.
+     * Calls {@link InputUtility#nextString(String, Predicate)}.
+     * Passes the prompt and a predicate that validates "y", "n", "yes", "no".
+     *
+     * @param prompt the prompt to display to the user.
+     * @return true for yes and false for no.
+     * @throws IOException when there's a problem with {@link InputStream}
+     */
+    public boolean nextYesNo(String prompt) throws IOException {
+        var result = this.nextString(prompt, yesOrNo()).trim().toLowerCase();
+        return YES_SYNONYMS.contains(result);
+    }
+
+    /**
+     * Calls {@link InputUtility#pause(String)}.
+     * The prompt defaults to "Press when ready".
+     *
+     * @throws IOException when there's a problem with {@link InputStream}
+     */
+    public void pause() throws IOException {
+        pause("Press when ready");
+    }
+
+    /**
+     * Calls nextString{@link InputUtility#nextString(String, Predicate)}.
+     * Uses prompt entered and waits for the user to press <enter>.
+     *
+     * @param prompt the prompt to display to the user.
+     * @throws IOException when there's a problem with {@link InputStream}
+     */
+    public void pause(String prompt) throws IOException {
+        this.nextString(prompt, String::isBlank);
+    }
+
+    /**
+     * Get a String predicate that tests for (case insensitive) "y", "n", "yes", "no".
+     *
+     * @return a String predicate that tests for (case insensitive) "y", "n", "yes", "no".
+     */
+    public static Predicate<String> yesOrNo() {
+        return s -> {
+            var cleanedString = s.trim().toLowerCase();
+            if (YES_SYNONYMS.contains(cleanedString)) {
+                return true;
+            }
+            else {
+                return NO_SYNONYMS.contains(cleanedString);
+            }
+        };
+    }
+
+    /**
+     * Get a String predicate that takes two or more Strings and tests whether any one of them match the input.
+     *
+     * @param these the strings for the input to match to.
+     * @return a String predicate that takes two or more Strings and tests whether any one of them match the input.
+     */
+    public static Predicate<String> oneOfThese(String... these) {
+        if (these.length < 2) {
```

Comment:
Why would you need at least two? One seems reasonable. Also, enforce a minimum number of arguments by detaching them from the varargs parameter:

`oneOfThese(String firstOption, String... otherOptions)`

Reply:
two was from the original requirements